project1/application.py

```diff
@@ -109,6 +110,67 @@ def logout():
    # remove the username from the session if it is there
    session.pop('username', None)
    return redirect(url_for('index'))
+   
+
+@app.route('/Search',methods=["POST"])
+def search():
+    print("searching")
+    search = request.form.get("search")
+    # isbn = request.form.get("isbn")
+    # title = request.form.get("book name")
+    # author = request.form.get("Author")
+    print(type(search))
+    ##  Using Like Operator 
+    if search != '' :
+        if request.form.get("isbn") == "option1":
+            print(search+" like  "+"isbn")
+            book_search = Books.query.filter(Books.isbn.like('%'+search+'%')).all()
+            #book_search = db1.session.query(Books).filter((Books.isbn.like('%'+search+'%')))
+            print(book_search)
+            type(book_search)
+            if (len(book_search) > 0):
+                return render_template("Search.html", books = book_search)
+            else :
+                return render_template("error.html", errors = "Sorry the details given doesnt match")
+           
+            # By title 
+        elif request.form.get("book name") == "option2":
+            print(search+" like   "+"book name")
+            book_search = db1.session.query(Books).filter((Books.tittle.like('%'+search+'%'))).all()
+            print(book_search)
+            if (len(book_search) > 0):
+                return render_template("Search.html", books = book_search)
+            else :
+                return render_template("error.html", errors = "Sorry the details given doesnt match")
+                
+        # user is searching by author name
+        elif request.form.get("Author") == "option3":
+            print(search+" like   "+"author")
+            book_search = db1.session.query(Books).filter((Books.author.like('%'+search+'%'))).all()
+            print(book_search)
```

Comment:
I see redundant lines can you reduce it and update.

Reply:
Were you able to fetch partial results also?

Reply:
Did you fix those redundant lines.

Reply:
Yes mam i had done it